kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ProduceResponse.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.entities.v3;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import java.time.Instant;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class ProduceResponse {
+
+  ProduceResponse() {
+  }
+
+  @JsonProperty("cluster_id")
+  public abstract String getClusterId();
+
+  @JsonProperty("topic_name")
+  public abstract String getTopicName();
+
+  @JsonProperty("partition_id")
+  public abstract int getPartitionId();
+
+  @JsonProperty("offset")
+  public abstract long getOffset();
+
+  @JsonProperty("timestamp")
+  @JsonInclude(Include.NON_ABSENT)
+  public abstract Optional<Instant> getTimestamp();
```

Comment:
wondering why this is optional

Reply:
My guess would be it's because this value more often than not comes from `RecordMetadata` [which is not always guaranteed to have a valid timestamp](https://github.com/apache/kafka/blob/a4c292173618e22b3f99ab8615c6b585c7688a40/clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java#L75-L81).

Reply:
Exactly. In practice we will always have a timestamp, since we always timestamp when producing. But since the Java API gives us a optional timestamp back, I have to propagate it, as there's no guarantees.